### PR TITLE
Migrate Uplink CLI to Metainfo API

### PR DESCRIPTION
**Pull request by kaloyan-raev**
The Uplink CLI is migrated to the new Metainfo API.

The Metainfo API is currently implemented on top of the existing storage implementation. Hence, the `streams.Store` is still used as a crutch. It will be removed at a later stage when the Metainfo API is fully reimplemented.

### CODE REVIEW

File: cmd/uplink/cmd/cp.go [hunk 1]
```diff
@@ -50,44 +49,67 @@ func upload(ctx context.Context, bs buckets.Store, src fpath.FPath, dst fpath.FP
...
 		return fmt.Errorf("source cannot be a directory: %s", src)
 	}
 
-	o, err := bs.GetObjectStore(ctx, dst.Bucket())
+	metainfo, streams, err := cfg.Metainfo(ctx)
+	if err != nil {
+		return err
+	}
+
+	bucket, err := metainfo.GetBucket(ctx, dst.Bucket())
```
Comment (egonelbre) [inline cmd/uplink/cmd/cp.go]: I think you should be able to use `metainfo.CreateObject` directly.

File: cmd/uplink/cmd/cp.go [hunk 2]
```diff
@@ -175,45 +202,61 @@ func copy(ctx context.Context, bs buckets.Store, src fpath.FPath, dst fpath.FPat
...
 
-	o, err := bs.GetObjectStore(ctx, src.Bucket())
+	metainfo, streams, err := cfg.Metainfo(ctx)
 	if err != nil {
-		return convertError(err, src)
+		return err
 	}
 
-	rr, _, err := o.Get(ctx, src.Path())
+	srcBucket, err := metainfo.GetBucket(ctx, src.Bucket())
```
Comment (egonelbre) [inline cmd/uplink/cmd/cp.go]: Same here, you should be able to use `metainfo.GetObjectStream` directly.

File: cmd/uplink/cmd/cp.go (hunk 1 above)
Comment (egonelbre) [reply cmd/uplink/cmd/cp.go]: Wait, didn't we add the "Bucket" info to mutable object or mutable stream?

Comment (kaloyan-raev) [reply cmd/uplink/cmd/cp.go]: We have just discussed it, but nothing done yet about this.

Comment (egonelbre) [reply cmd/uplink/cmd/cp.go]: ah, ok, let's leave as is then

**Pull request merged**